src/firefly/js/api/ApiViewer.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import {has, isArray} from 'lodash';
+import {debug} from './ApiUtil.js';
+import {getRootURL}  from '../util/BrowserUtil.js';
+import {getCookie}  from '../util/WebUtil.js';
+import {aliveCheck, dispatchRemoteAction}  from '../rpc/PushServices.js';
+import {dispatchPlotImage}  from '../visualize/ImagePlotCntlr.js';
+import {RequestType}  from '../visualize/RequestType.js';
+import {clone}  from '../util/WebUtil.js';
+import {confirmPlotRequest,findInvalidWPRKeys}  from '../visualize/WebPlotRequest.js';
+
+
+/**
+ * Build the interface to remotely communicate to the firefly viewer
+ * @return {{getViewer: getViewer, getExternalViewer: getExternalViewer}}
+ */
+export function buildViewerApi() {
+    return {getViewer,getExternalViewer};
+}
+
+function getDefChannel() {
+    var root= getCookie('usrkey');
+    if (root) {
+        const idx= root.indexOf('/');
+        if (idx>-1) root= root.substring(0,idx);
+    }
+    else {
+        root= 'remote';
+    }
+    return `${root}-viewer`;
+}
```

Comment:
not sure if you are aware, WebSocketClient capture the channel information and stored the connection id and channel in a 'seinfo' cookie.  it would be trivial to expose those information through WebSocketClient.